tests/integration/test_external_agent.py

```diff
@@ -0,0 +1,57 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Integration tests for jenkins-k8s-operator with ingress."""
+
+import typing
+
+import pytest
+from juju.application import Application
+
+import state
+from charm import AGENT_DISCOVERY_INGRESS_RELATION_NAME
+
+
+# This will only work on microk8s !!
```

Comment:
I can't remember, I think we somewhere have implemented a check to see if this is running on microk8s?

Reply:
iirc, ops_test will automatically fail automatically :D

Reply:
I just moved this test  from `test_ingress.py` to a new test file as sharing a traefik instance seemed too unstable (not from the charm's point of view but from ops_test's POV) and that module was failing quite frequently. Splitting that module into 2 test files seemed to have solved the instability issue.

This test is the only test where it only works on microk8s, and in its original PR [I mentioned that](https://github.com/canonical/jenkins-k8s-operator/pull/115/commits/2c00c97686066481356a17eec49732f0fb1e59ad) I couldn't find a good way to test for it, so I'll only keep a comment here for now.